### PR TITLE
Adds Scala default files

### PR DESCRIPTION
**Pull request by juanpedromoreno**
This PR adds the default files along with a CI workflow for checking them.

### CODE REVIEW

File: .github/workflows/release-defaults.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+          - os: ubuntu-latest
+            file_name: .scalafmt.conf
+
+    steps:
+    - uses: hecrj/setup-rust-action@v1-release
+      with:
+        rust-version: stable
+    - uses: actions/checkout@v1
+    - name: Build
+      run: cargo build --release --locked
```
Comment (BenFradet) [inline .github/workflows/release-defaults.yml:26]: what's the use of rust?

Comment (juanpedromoreno) [reply .github/workflows/release-defaults.yml:26]: It's part of the github action I'm using here to upload the default files to the github release: https://github.com/marketplace/actions/upload-files-to-a-github-release

Comment (BenFradet) [reply .github/workflows/release-defaults.yml:26]: so will this create scalafix and scalafmt files as release assets to this repo?

Comment (juanpedromoreno) [reply .github/workflows/release-defaults.yml:26]: Yes, and with sbt-fix, we can use them with `scalafixConfigLocation` and `scalafmtConfigLocation` settings.

File: .github/workflows/check-defaults.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout project
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```
Comment (alejandrohdezma) [inline .github/workflows/check-defaults.yml:15]: Why do you need to download the whole history?

File: .github/workflows/check-defaults.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      - name: Fetch tags
+        run: git fetch --tags
```
Comment (alejandrohdezma) [inline .github/workflows/check-defaults.yml:17]: Why do you need to download repo tags?

File: .github/workflows/release-defaults.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+on:
+  push:
+    tags:
+      - '*'
+
+jobs:
+  publish:
+    name: Publish ${{ matrix.file_name }}
+    runs-on: ${{ matrix.file_name }}
```
Comment (alejandrohdezma) [inline .github/workflows/release-defaults.yml:11]: Shouldn't this be `matrix.os`?

```suggestion
    runs-on: ${{ matrix.os }}
```

File: .github/workflows/release-defaults.yml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+          - os: ubuntu-latest
+            file_name: .scalafix.conf
+          - os: ubuntu-latest
+            file_name: .scalafmt.conf
+
+    steps:
+    - uses: hecrj/setup-rust-action@v1-release
+      with:
+        rust-version: stable
+    - uses: actions/checkout@v1
```
Comment (alejandrohdezma) [inline .github/workflows/release-defaults.yml:24]: Use `actions/checkout@v2` if possible

```suggestion
    - uses: actions/checkout@v2
```

File: .github/workflows/release-defaults.yml [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-latest
+            file_name: .scalafix.conf
+          - os: ubuntu-latest
+            file_name: .scalafmt.conf
+
+    steps:
+    - uses: hecrj/setup-rust-action@v1-release
```
Comment (alejandrohdezma) [inline .github/workflows/release-defaults.yml:21]: I don't understand why do you need to install rust or use `cargo build` here if you are just uploading two files already checked out.

File: .github/workflows/release-defaults.yml [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+  push:
+    tags:
+      - '*'
+
+jobs:
+  publish:
+    name: Publish ${{ matrix.file_name }}
+    runs-on: ${{ matrix.file_name }}
+    strategy:
+      matrix:
```
Comment (alejandrohdezma) [inline .github/workflows/release-defaults.yml:13]: The matrix can be simplified with only the `file_name` (given the os is the same for the two files)

File: root-files/.gitignore [hunk 8]
```diff
@@ -2,6 +2,7 @@
 
 .idea
 out/
+scalafmt
```
Comment (alejandrohdezma) [inline root-files/.gitignore:5]: This shouldn't go here, it should go in this repo's `.gitignore`

File: .github/workflows/release-defaults.yml (hunk 6 above)
Comment (juanpedromoreno) [reply .github/workflows/release-defaults.yml:21]: I just followed the tutorial, I can try deleting it 👍

**Comment by juanpedromoreno** [comment]
Comments addressed @alejandrohdezma , could you take another look? Thanks.

**Comment by alejandrohdezma** [comment]
Great work!!

**Pull request merged**